ovs/dal/hybrids/albabackend.py

```diff
@@ -42,105 +43,145 @@ class AlbaBackend(DataObject):
+                for disk_entry in _node_data.values():
+                    for entry in disk_entry['asds'].values():
+                        entry['status_detail'] = 'nodedown'
+                _asd_data = {}
+            for _disk_id, asds in _asd_data.iteritems():
+                if _disk_id not in _node_data:
+                    continue
+                for _asd_id, asd_info in asds.iteritems():
+                    entry = {'asd_id': _asd_id,
+                             'status': 'error' if asd_info['state'] == 'error' else 'initialized',
+                             'status_detail': asd_info.get('state_detail', ''),
+                             'state': asd_info['state'],
+                             'state_detail': asd_info.get('state_detail', '')}
+                    if _asd_id not in _node_data[_disk_id]['asds']:
+                        _node_data[_disk_id]['asds'][_asd_id] = entry
+                    else:
+                        _node_data[_disk_id]['asds'][_asd_id].update(entry)
         threads = []
         for node in alba_nodes:
-            thread = Thread(target=_load_disks, args=(node, node_disk_map[node.node_id]))
+            thread = Thread(target=_load_live_info, args=(node, storage_map[node.node_id]))
             thread.start()
             threads.append(thread)
         for thread in threads:
             thread.join()
 
-        # Make mapping between node IDs and the relevant OSDs and disks
+        # Load information from alba
         backend_interval_key = '/ovs/alba/backends/{0}/gui_error_interval'.format(self.guid)
         if EtcdConfiguration.exists(backend_interval_key):
             interval = EtcdConfiguration.get(backend_interval_key)
         else:
             interval = EtcdConfiguration.get('/ovs/alba/backends/global_gui_error_interval')
-        disks = []
-        for node in alba_nodes:
-            for info in node_disk_map[node.node_id]:
-                disk = info.get('disk')
-                if disk is None:
-                    continue
-                disk_status = 'uninitialized'
-                disk_status_detail = ''
-                disk_alba_backend_guid = ''
-                if disk['available'] is False:
-                    osd = info.get('osd')
-                    disk_alba_state = disk['state']['state']
-                    if disk_alba_state == 'ok':
-                        if osd is None:
-                            disk_status = 'initialized'
-                        elif osd['id'] is None:
-                            alba_id = osd['alba_id']
-                            if alba_id is None:
-                                disk_status = 'available'
+        config = 'etcd://127.0.0.1:2379/ovs/arakoon/{0}/config'.format(self.abm_services[0].service.name)
+        for found_osd in AlbaCLI.run('list-all-osds', config=config, as_json=True):
+            node_id = found_osd['node_id']
+            asd_id = found_osd['long_id']
+            if node_id in storage_map:
```

Comment:
Amount of indents can be diminished
Eg:
```if node_id in storage_map:
    for _disk in storage_map[node_id].values():```
can become
```for _disk in storage_map.get(node_id, {}).values():```

And on line 148
```if asd_id in _disk['asds']:
    asd_data = _disk['asds'][asd_id]```
can become
```asd_data = _disk['asds'].get(asd_id, {})```